app/bundles/EmailBundle/Tests/MonitoredEmail/Accessor/ConfigAccessorTest.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * @copyright   2017 Mautic Contributors. All rights reserved
+ * @author      Mautic, Inc.
+ *
+ * @link        https://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\EmailBundle\Tests\MonitoredEmail\Accessor;
+
+use Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor;
+
+class ConfigAccessorTest extends \PHPUnit_Framework_TestCase
+{
+    protected $config = [
+        'imap_path' => 'path',
+        'user'      => 'user',
+        'host'      => 'host',
+        'folder'    => 'folder',
+    ];
+
+    /**
+     * @testdox All getters return appropriate values
+     *
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getPath()
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getUser()
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getHost()
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getFolder()
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getProperty()
+     */
+    public function testGetters()
+    {
+        $configAccessor = new ConfigAccessor($this->config);
+
+        $this->assertEquals($this->config['imap_path'], $configAccessor->getPath());
+        $this->assertEquals($this->config['user'], $configAccessor->getUser());
+        $this->assertEquals($this->config['host'], $configAccessor->getHost());
+        $this->assertEquals($this->config['folder'], $configAccessor->getFolder());
+    }
+
+    /**
+     * @testdox Key is formatted appropriately
+     *
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getKey()
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getHost()
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getFolder()
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getProperty()
+     */
+    public function testKeyIsPathAndUser()
+    {
+        $configAccessor = new ConfigAccessor($this->config);
+
+        $this->assertEquals('path_user', $configAccessor->getKey());
+    }
+
+    /**
+     * @testdox Test its considered configured if we have a host and a folder
+     *
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::isConfigured()
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getHost()
+     * @covers \Mautic\EmailBundle\MonitoredEmail\Accessor\ConfigAccessor::getFolder()
```

Comment:
Does this test really cover all these methods?

Reply:
It does because isConfigured() uses getHost() and getFolder() and so the test would fail and any of those three are false.